### ISSUE TITLE
default_numeric_fallback should not warn code generated by external macros

### ISSUE THREAD

**Issue by taiki-e** [rust-lang/rust-clippy#7304]
I tried this code:

```rust
#[warn(clippy::default_numeric_fallback)]
pub async fn myfunc() {
    tokio::select! {
        _ = async {} => (),
        else => (),
    }
}
```

I expected to see this happen: no warning

Instead, this happened: default_numeric_fallback warns code generated by external macros

```text
warning: default numeric fallback might occur
 --> src/lib.rs:3:5
  |
3 | /     tokio::select! {
4 | |         _ = async {} => (),
5 | |         else => (),
6 | |     }
  | |_____^
  |
note: the lint level is defined here
 --> src/lib.rs:1:8
  |
1 | #[warn(clippy::default_numeric_fallback)]
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#default_numeric_fallback
  = note: this warning originates in the macro `$crate::count` (in Nightly builds, run with -Z macro-backtrace for more info)
```

[playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=3820c9330317d7ed55ca4c459469a428)

*This originally reported by @kamulos in https://github.com/tokio-rs/tokio/issues/3830.*

### Meta

- `cargo clippy -V`: e.g. clippy 0.1.54 (c1e8f3a5 2021-05-30)
- `rustc -Vv`:
  ```
  rustc 1.54.0-nightly (c1e8f3a58 2021-05-30)
  binary: rustc
  commit-hash: c1e8f3a5857c3c8d095fd212acc6ed442f4cc58b
  commit-date: 2021-05-30
  host: x86_64-apple-darwin
  release: 1.54.0-nightly
  LLVM version: 12.0.1
  ```

**Comment by taiki-e** [comment]
Mentioning @Y-Nak who implemented this lint in #6662.

**Comment by Y-Nak** [comment]
@rustbot claim
I'll fix this in this week.
